### PR TITLE
Gov type option

### PR DESCRIPTION
**Pull request by Sarahtonein**
**Whats new**
- Removed textInput field for proposal type
- Created a button selection option instead
- Removed now redundant error handling of proposal type

This removes the need for:

a - A user to somehow magically *know* the proposal types
b - Error handling / frustration of the wrong type being provided

**How to test**
- docker-compose up --build
- invoke /vote_draft command
- create a proposal

image

image

### CODE REVIEW

**Comment by Sarahtonein** [comment]
Updated proposal type buttons as a separate ephemeral message as it was previously displaying the same text from the initial invocation

image

File: README.md [hunk 1]
```diff
@@ -122,9 +122,9 @@ All commands use discords slash commands (/)
 
 ```/vote_draft``` 
 
-Upon typing the command the bot will present you with three buttons: Create, Edit, Delete.
+Upon typing the command the bot will present you with three buttons: Create, Edit, Preview, and Delete. 
 
-These should be self explained, create will allow you to draft a proposal. Edit will allow you to edit an existing proposal, and delete
+These should be self explained, clicking create will prompt you with either general or budget, these options dictate the type of proposal. Edit will allow you to edit an existing proposal, and delete
```
Comment (bagelface) [inline README.md:127]:
```
"Create" will create a new proposal. Note: "Proposal Type" field should be either "General" or "Budget".
"Edit" will allow you to edit an existing proposal.
"Preview" will allow you to preview an existing proposal before it is published.
"Delete" will let you delete and existing proposal that has not been published.
```

File: README.md [hunk 2]
```diff
@@ -122,9 +122,9 @@ All commands use discords slash commands (/)
 
 ```/vote_draft``` 
 
-Upon typing the command the bot will present you with three buttons: Create, Edit, Delete.
+Upon typing the command the bot will present you with three buttons: Create, Edit, Preview, and Delete.
```
Comment (bagelface) [inline README.md:125]: Upon typing the command, the bot will present you with four buttons: "Create", "Edit", "Preview", and "Delete".

File: proposals/proposal_modal.py [hunk 3]
```diff
@@ -39,14 +34,14 @@ class ProposalModal(ui.Modal, title="Create/Edit Proposal"):
         max_length=2000,
     )
 
-    def __init__(self, channel, proposal):
+    def __init__(self, channel, proposal, proposal_type=None):
```
Comment (bagelface) [inline proposals/proposal_modal.py:37]: does this need to have a "None" default value or can it just be a required field?

File: proposals/proposal_buttons_view.py
Comment (bagelface) [inline proposals/proposal_buttons_view.py:1]: I think adding "Proposal" to the buttons is too wordy. It's implied they're all related to proposals

File: proposals/proposal_buttons_view.py [hunk 4]
```diff
@@ -4,20 +4,31 @@
...
-    @discord.ui.button(label="Create", style=discord.ButtonStyle.green)
-    async def create(self, interaction: discord.Interaction, button: discord.ui.Button):
-        modal = ProposalModal(interaction.channel, None)
-        await interaction.response.send_modal(modal)
+    @discord.ui.button(label="Create Proposal", style=discord.ButtonStyle.green)
+    async def create_proposal(
+        self, interaction: discord.Interaction, button: discord.ui.Button
+    ):
+        view = discord.ui.View()
+        view.add_item(CreateGeneralProposalButton(self.proposals))
```
Comment (bagelface) [inline proposals/proposal_buttons_view.py:24]: I don't have this set up to run locally, can you provide a screenshot of what this looks like?

**Review by bagelface — CHANGES_REQUESTED** [review]
a few suggestions

File: README.md (hunk 1 above)
Comment (Sarahtonein) [reply README.md:127]: Sounds good

> I think this would be better written like this (no need to explain the "general" "budget" selection):
> 
> ```
> "Create" will create a new proposal.
> "Edit" will allow you to select an an existing proposal to edit.
> "Preview" will allow you to preview an existing proposal before it is published.
> "Delete" will allow you delete and existing proposal that has not been published.
> ```

File: proposals/proposal_buttons_view.py (hunk 4 above)
Comment (Sarahtonein) [reply proposals/proposal_buttons_view.py:24]: > I don't have this set up to run locally, can you provide a screenshot of what this looks like?

image

File: proposals/proposal_modal.py (hunk 3 above)
Comment (Sarahtonein) [reply proposals/proposal_modal.py:36]: > does this need to have a "None" default value or can it just be a required field?

Didn't need to be none and noticed an issue here when editing a proposal, I've updated this now to address that problem. Thanks for picking this up

**Comment by Sarahtonein** [comment]
> a few suggestions

Addressed :)

**Comment by bagelface** [comment]
lgtm
